lib/routes/github/private-feed.ts

```diff
@@ -0,0 +1,230 @@
+            description = `Created ${payload.ref_type}: ${payload.ref || repo.name}`;
+            link = `https://github.com/${repo.name}`;
+            break;
+        case 'DeleteEvent':
+            title = `${actor.login} deleted ${payload.ref_type} in ${repo.name}`;
+            description = `Deleted ${payload.ref_type}: ${payload.ref}`;
+            link = `https://github.com/${repo.name}`;
+            break;
+        case 'ReleaseEvent':
+            title = `${actor.login} released ${payload.release?.name || payload.release?.tag_name} in ${repo.name}`;
+            description = payload.release?.body || `Released ${payload.release?.tag_name}`;
+            link = payload.release?.html_url || `https://github.com/${repo.name}`;
+            break;
+        case 'PublicEvent':
+            title = `${actor.login} made ${repo.name} public`;
+            description = `Repository ${repo.name} was made public`;
+            link = `https://github.com/${repo.name}`;
+            break;
+        case 'MemberEvent':
+            title = `${actor.login} ${payload.action} as a member of ${repo.name}`;
+            description = `Member ${payload.action} in repository ${repo.name}`;
+            link = `https://github.com/${repo.name}`;
+            break;
+        default:
+            title = `${actor.login} performed ${type} in ${repo?.name || 'unknown repository'}`;
+            description = `Activity type: ${type}`;
+            link = repo ? `https://github.com/${repo.name}` : `https://github.com/${actor.login}`;
+    }
+
+    return {
+        title,
+        link,
+        description,
+        pubDate: parseDate(created_at),
+        author: actor.login,
+        category: [type],
+    };
+}
+
+async function handler(ctx) {
+    const user = ctx.req.param('user');
+    const types = ctx.req.param('types') || 'all';
+
+    // Setup headers with optional authentication
+    const headers: Record<string, string> = {};
+    if (config.github && config.github.access_token) {
+        headers.Authorization = `token ${config.github.access_token}`;
+    }
+
+    const response = await got({
+        method: 'get',
+        url: `https://api.github.com/users/${user}/received_events`,
+        headers,
+        searchParams: {
+            per_page: 30,
+        },
+    });
+
+    // Parse requested event types and map short names to full event types
+    let filteredEventTypes: string[] = [];
+    if (types !== 'all') {
+        filteredEventTypes = types.split(',').map((type) => {
+            const trimmedType = type.trim();
+            return typeMapping[trimmedType] || trimmedType;
+        });
+    }
+
+    const items = response.data
+        .filter((event) => filteredEventTypes.length === 0 || filteredEventTypes.includes(event.type))
+        .map((event) => formatEventItem(event));
+
+    const typeFilter = types === 'all' ? 'All Events' : `Events: ${types}`;
+    const isAuthenticated = config.github && config.github.access_token;
```

Comment:
Define `isAuthenticated` before https://github.com/DIYgod/RSSHub/blob/e489ca70449345d977c799f1770e9f1fcf43f040/lib/routes/github/private-feed.ts#L194 and use it in the `if`